exec/docker_swarm_stack_status.go

```diff
@@ -0,0 +1,170 @@
+package exec
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/client"
+	"github.com/portainer/agent/docker"
+	libstack "github.com/portainer/portainer/pkg/libstack"
+	"github.com/rs/zerolog/log"
+)
+
+func (service *DockerSwarmStackService) WaitForStatus(ctx context.Context, name string, status libstack.Status) <-chan string {
+	errorMessageCh := make(chan string)
+
+	go func() {
+		timeout := time.After(5 * time.Minute)
```

Comment:
Why do we need the timeout if we (can) have it in the context?

Reply:
right, it's left from before I added timeout to context